### PR TITLE
Add USE_KUDU_NESTED_JOIN hint

### PR DESCRIPTION
**Pull request by twdsilva**
Add a hint that can be used to force a query to use KuduNestedJoin (by default the EnumerableHashJoin is used). 

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java [hunk 1]
```diff
@@ -81,24 +87,32 @@ public Boolean visitInputRef(final RexInputRef inputRef) {
...
     };
 
-    final Boolean isValid = condition.accept(validateJoinCondition);
-
-    // @TODO: Hack for the moment.
-    final boolean containsActorSid = join.getRight().getRowType().getFieldList().stream()
-        .anyMatch(fl -> fl.getName().equalsIgnoreCase("actor_sid"));
-
-    return isValid != null && isValid && containsActorSid;
+    return condition.accept(validateJoinCondition);
```
Comment (sdreynolds) [inline adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java:90]: `validateJoinCondition` can return `null`. I believe you need to check for null

**Review by sdreynolds — APPROVED** [review]
Please add a check for null and run `mvn spotless:apply` to fix formats then merge

File: adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java [hunk 2]
```diff
@@ -30,28 +30,34 @@
...
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+
 public class KuduNestedJoinRule extends RelOptRule {
   public final static EnumSet<SqlKind> VALID_CALL_TYPES = EnumSet.of(SqlKind.EQUALS, SqlKind.GREATER_THAN,
       SqlKind.GREATER_THAN_OR_EQUAL, SqlKind.LESS_THAN, SqlKind.LESS_THAN_OR_EQUAL);
 
-  public static final int DEFAULT_BATCH_SIZE = 100;
+  public static final int DEFAULT_BATCH_SIZE = 5;
```
Comment (rbhanot) [inline adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java:41]: Curious - Any reason to change the batch size ?

File: adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java (hunk 1 above)
Comment (twdsilva) [reply adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java]: done

File: adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java (hunk 2 above)
Comment (twdsilva) [reply adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java]: was just for testing reverted

**Pull request merged**